### PR TITLE
robot: fix error on unknown tests [v2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
- Use the specific exception.
- Bonus commit: pylint fixes

v1: #1889

### CODE REVIEW

File: optional_plugins/robot/avocado_robot/__init__.py [hunk 1]
```diff
@@ -17,8 +17,6 @@
 import re
 import subprocess
 
-from StringIO import StringIO
-
```
Comment (ldoktor) [inline optional_plugins/robot/avocado_robot/__init__.py]: Nice bonus commit, but when you are on the cleanup, you should fix all or none of the pylint issues 😉

```
C0111:   1,0: Missing module docstring
R0913:  40,4: Too many arguments (6/5)
W0613:  45,17: Unused argument 'external_runner'
W0702:  97,8: No exception type(s) specified
W0621: 112,12: Redefining name 'test' from outer scope (line 25)
...
Global evaluation
-----------------
Your code has been rated at 8.97/10 (previous run: 8.97/10, +0.00)
```

What do you say?

Comment (ldoktor) [inline optional_plugins/robot/avocado_robot/__init__.py]: (I mean either you are doing import clenaups on all files, or you do pylint cleanup on a single file, but doing just one minor fix on a single file seems odd to me)

Comment (apahim) [inline optional_plugins/robot/avocado_robot/__init__.py]: Sure. Updating.

**Comment by apahim** [comment]
@ldoktor, updated.

**Comment by ldoktor** [comment]
Very nice, how about one extra change:

```diff
diff --git a/optional_plugins/robot/avocado_robot/__init__.py b/optional_plugins/robot/avocado_robot/__init__.py
index 6f495b0..5eb117d 100644
--- a/optional_plugins/robot/avocado_robot/__init__.py
+++ b/optional_plugins/robot/avocado_robot/__init__.py
@@ -39,8 +39,7 @@ class RobotTest(test.SimpleTest):
                  name,
                  params=None,
                  base_logdir=None,
-                 job=None,
-                 external_runner=None):
+                 job=None):
         super(RobotTest, self).__init__(name, params, base_logdir, job)
 
     @property
```

and please (merge) and rebase it on top of https://github.com/avocado-framework/avocado/pull/1894

**Pull request closed without merging**